libff/algebra/curves/profile/algebra_groups_read_profile.cpp

```diff
@@ -75,40 +90,568 @@ bool profile_group_read_sequential_uncompressed(const std::string &identifier)
+            // std::cout << "cb_wait: done\n";
+            return ret;
+        }
+
+        if (err == ECANCELED) {
+            throw std::runtime_error("aio_error: cancelled");
+        }
+
+        throw std::runtime_error(
+            std::string("error from aio_error: ") + strerror(errno));
+    }
 }
 
-template<typename GroupT> void run_profile(const std::string &identifier)
+template<size_t BatchSize> class batched_aio_reader
 {
-    std::cout << " profile: " << identifier << "\n";
+public:
+    batched_aio_reader(int fd)
+        : _fd(fd)
+        , _active_batch(_batch1_ptrs)
+        , _next_batch(_batch2_ptrs)
+        , _next_batch_size(0)
+    {
+        memset(_batch1, 0, sizeof(_batch1));
+        memset(_batch2, 0, sizeof(_batch2));
+        for (size_t i = 0; i < BatchSize; ++i) {
+            _batch1_ptrs[i] = &_batch1[i];
+            _batch2_ptrs[i] = &_batch2[i];
+        }
+    }
+
+    void enqueue_read_first_batch(
+        size_t offset_bytes, size_t size_bytes, void *dest)
+    {
+        assert(_next_batch_size < BatchSize);
+
+        aiocb *cb = _next_batch[_next_batch_size];
+        cb_init(cb, _fd, offset_bytes, size_bytes, dest);
+
+        // When first batch is full, enqueue it and allow writing to the next
+        // batch.
+
+        ++_next_batch_size;
+        if (_next_batch_size == BatchSize) {
+            enqueue_next_batch();
+
+            std::swap(_active_batch, _next_batch);
+            _next_batch_size = 0;
+        }
+    }
+
+    void enqueue_read(size_t offset_bytes, size_t size_bytes, void *dest)
+    {
+        assert(_next_batch_size < BatchSize);
+
+        aiocb *cb = _next_batch[_next_batch_size];
+        cb_init(cb, _fd, offset_bytes, size_bytes, dest);
+
+        ++_next_batch_size;
+        if (_next_batch_size == BatchSize) {
+            enqueue_next_batch();
+        }
+    }
+
+    void wait_last_read()
+    {
+        // Wait for the _active_batch
+        for (size_t i = 0; i < BatchSize; ++i) {
+            int r = cb_wait(_active_batch[i]);
+            if (0 > r) {
+                throw std::runtime_error("bad read result");
+            }
+            if (0 == r) {
+                std::cout << "+";
+            }
+        }
+
+        // swap pointers so it can be written into
+        std::swap(_active_batch, _next_batch);
+        _next_batch_size = 0;
+    }
+
+protected:
+    void enqueue_next_batch()
+    {
+        int r = lio_listio(LIO_NOWAIT, _next_batch, BatchSize, nullptr);
+        if (r != 0) {
+            throw std::runtime_error("enqueue_batch error");
+        }
+    }
+
+    const int _fd;
+    aiocb _batch1[BatchSize];
+    aiocb _batch2[BatchSize];
+    aiocb *_batch1_ptrs[BatchSize];
+    aiocb *_batch2_ptrs[BatchSize];
+    aiocb **_active_batch; // Next batch to wait for
+    aiocb **_next_batch;   // Batch being filled
+    size_t _next_batch_size;
+};
+
+/// Similar to profile_group_read_random_aio_ordered_uncompressed, but with
```

Comment:
Following this thread: https://github.com/clearmatics/libff/pull/46#discussion_r690345919

If the function has been removed, this comment needs to be modified since the function referenced does not seem to exist anymore.

Reply:
Reworded the comment.  Thanks.